components/tidb_query/src/codec/datum.rs

```diff
@@ -233,7 +240,8 @@ impl Datum {
             }
             Datum::Time(ref t) => Ok(t.cmp(time)),
             _ => {
-                let f = time.convert(ctx)?;
```

Comment:
I think we should not make this change. (we can infer this logic is correct even though the code is not exactly same as TiDB)